pkg/controllers/ttl/controller.go

```diff
@@ -117,6 +117,31 @@ func (c *controller) deregisterEventHandlers() {
 	c.logger.V(3).Info("deregistered event handlers")
 }
 
+// Function to determine the deletion propagation policy
+func (c *controller) determinePropagationPolicy(metaObj metav1.Object, logger logr.Logger) *metav1.DeletionPropagation {
+	annotations := metaObj.GetAnnotations()
+	var policy *metav1.DeletionPropagation
+
+	if annotations != nil {
+		if annotationPolicy, ok := annotations["propagationPolicy"]; ok {
```

Comment:
Use a better annotation name (we discussed it in out weekly meeting).

Reply:
Sure

Reply:
hey @eddycharly! Should I go with this annotation name "cleanupPropagationPolicy" or something else

Reply:
Please use a name similar to https://github.com/kyverno/kyverno/blob/7763ff72cf0303ddc2f169300003c78888bd6373/api/kyverno/constants.go#L9 (and declare it in this file)

Reply:
Okay thanks for the guidance @eddycharly